drivers/adc/adc_stm32.c

```diff
@@ -736,13 +736,6 @@ static int start_read(const struct device *dev,
 		LL_ADC_SetChannelPreselection(adc, channel);
 	}
 #endif
-#if defined(CONFIG_SOC_SERIES_STM32U5X)
-	/*
-	 * Each channel in the sequence must be previously enabled in PCSEL.
-	 * This register controls the analog switch integrated in the IO level.
-	 */
-	LL_ADC_SetChannelPreselection(adc, channel);
```

Comment:
Is it the same code? The one at 736 is only for adc1. I understand it may be functionally equivalent but then maybe mention it in the patch comment.

Reply:
This preselection has been discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/48770
Based on the reference manual RM0456, ADC4 doesn't have PCSEL register, so the version with a check is the better one.